src/app/modules/browse/browse-controller.js

```diff
@@ -83,23 +83,38 @@
                 params.bookOrBooks = _localize('book');
             }
 
+            var isFiltered = false;
+
             if (count === 0) {
-                message = _localize('There are no books that match your search for {shelf} {language} books', params);
+                $scope.resultsMessage = _localize('There are no books that match your search', params);
```

Comment:
Three reasons we thought this might be better:
1) simplified code
2) reduced redundancy for the translators
3) scalability

The previous message attempted to put the whole thing in an English sentence.  There were 8 versions of the message:
There are no books that match your search for {shelf} {language} books
There are no books that match your search for {shelf} {language} books with the {tag} tag
There are no books that match your search for {shelf} {language} books with the {tag} tag containing "{searchText}"
There are no books that match your search for {shelf} {language} books containing "{searchText}"
Found {count} {shelf} {language} {bookOrBooks}
Found {count} {shelf} {language} {bookOrBooks} with the {tag} tag
Found {count} {shelf} {language} {bookOrBooks} with the {tag} tag containing "{searchText}"
Found {count} {shelf} {language} {bookOrBooks} containing "{searchText}"

The only way to properly internationalize these is to have 8 distinct strings in the code and 8 distinct strings (with a lot of duplication) for the translator.

For scalability, if we add more ways the search can be filtered, it may become impossible to reasonably form English sentences.

Of course, it is your call.